etl/bulk_plot.py

```diff
@@ -0,0 +1,547 @@
+"""HMS Flow Analysis Module.
+
+This module provides functionality for querying and plotting all HMS
+flow data from FFRD S3 storage, including AMS (Annual Maximum Series)
+peaks and confidence limits.
+"""
+
+from __future__ import annotations
+
+import logging
+import os
+from pathlib import Path
+
+import duckdb
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import pandas as pd
+import s3fs
+from dotenv import load_dotenv
+
+__all__ = ["plot_all_hms_elements"]
+
+_ = load_dotenv()
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+
+
+def _create_s3_connection(aws_region: str = "us-east-1") -> duckdb.DuckDBPyConnection:
```

Comment:
Can we please move this to the s3_utils.py and also look at functions in the sibling files that should go there? For example `s3_output_exists` in ams_query could be moved as well.